modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala

```diff
@@ -314,6 +316,58 @@ object EpicsUtil {
 
   def waitForValue[T](attr: CaAttribute[T], v: T, timeout: Time, name: String): SeqAction[Unit] = waitForValues[T](attr, List(v), timeout, name).void
 
+  def waitForValuesF[T, F[_]: Async](attr: CaAttribute[T], vv: Seq[T], timeout: Time, name: String): F[T] =
+    Async[F].async[T] { (f: Either[Throwable, T] => Unit) =>
+      //The task is created with async. So we do whatever we need to do,
+      // and then call `f` to signal the completion of the task.
+
+      //`resultGuard` and `lock` are used for synchronization.
+      val resultGuard = new AtomicInteger(1)
```

Comment:
We don't need to do this for this PR but you may consider to check the concurrent utilities from cats
https://typelevel.org/cats-effect/concurrency/mvar.html